pynbody/snapshot/tipsy.py

```diff
@@ -1181,61 +1169,115 @@ def __init__(self, filename, sort=True, paramfile=None):
             size = struct.unpack(">i", f.read(4))
         iSize = size[0]
 
-        if (iSize > file_structure.itemsize):
+        bigstarlog = False
+        molecH = False
+        bigIOrds = False
+
+
+        if use_log:
+            # assumes log file would be in the same location as starlog file
+            logger.info('Attempting to load starlog metadata from log file')
+            self._logfile = filename.replace('starlog', 'log')
```

Comment:
This looks dangerous in principle, since someone might have `starlog` as part of their file or path name

Perhap, ahead of `if use_log`, something like

```python
self._logfile, n_replacements = re.subn(r"\.starlog$", ".log", filename)
if(use_log and n_replacements != 1):
    warnings.warn('Unable to determine log filename; reverting to guess-and-check')
    use_log = False
```

(I didn't test this)

Reply:
I have changed it so only the last `starlog` is replaced, which should be safe, given that earlier searches assume the starlog file is the file that ends in `*.starlog`. If you still prefer I use your approach, I'm happy to change it.

I have also added two starlog reading tests - one with and one without a log file. Let me know if this is what you had in mind.